content/2017.12.kinto-at-mozilla.rst

```diff
@@ -0,0 +1,148 @@
+
+In order to manage buckets and collections, as well as groups and their related permissions, we use `kinto-wizard` that reads a YAML file from a Github private repo, and creates or updates the missing objects.
+
+Instance: https://firefox.settings.services.mozilla.com
+Production date: December 2015
+Average requests/sec: 10 (*behind the Cloudfront CDN*)
+
+Blocklists
+----------
+
+The blocklists allow to remotely disable malware addons, malicious or unstable plugins, or some graphical drivers that come to be unstable on certain hardware. A custom plugin, kinto-amo, was implemented to serve the blocklist in the same XML format as addons.mozilla.org used to have — for old Firefox versions.
+
+Certificates revocation
+-----------------------
+
+Certicates sometimes need to be invalidated before they expire. The OneCRL (certificates revocation list) fulfils that goal.
+
+Certificates pinning
+--------------------
+
+When the browser reaches a website via https for the first time, it downloads its certificate. To make sure the certificate is genuine, we use a list of public keys (HPKP) to be compared on the client side.
+
+Android assets
+--------------
+
+In order to reduce the size of the Android installer, some assets — like fonts or hyphenations dictionnaries — are downloaded asynchronously. The client polls files from *Firefox remote settings*, leveraging `kinto-attachment`_.
+
+Android and iOS experiments
+---------------------------
+
+This stack is also used to deliver A/B testing features on Android and iOS. The client receives metadata for the on-going experiments.
+
+Resist fingerprinting with fonts whitelist
+------------------------------------------
+
+`The idea <https://bugzilla.mozilla.org/show_bug.cgi?id=1336208>`_ is to deliver some fonts to avoid downloading them while browsing. The client globally works like the Android assets.
+
+Possible evolutions
+-------------------
+
+- There is certainly going to be more types of settings for new use-cases
+- The stack currently uses `kinto-ldap` to authenticate users. We'll switch to OpenID Connect now that the company uses Auth0 everywhere.
+- The `Push team` is working on a solution to replace frequent polling with a push (at that scale nothing is trivial).
+- The Sync team is working on a new data store for profile data. We'll probably replace the current SQlite adapter.
+
+
+WebExtensions storage.sync API
+==============================
+
+The tough one!
+
+Firefox is now compatible with the Web Extensions APIs, that allow addons developers to share a common code-base between different browsers. One of these APIs allows addons to store and sync some data (eg. user-preferences).
+In our implementation, the client encrypts the data before sending it to our servers to guarantee privacy. It might not be the case of Chrome by the way. It uses `kinto.js` and leverages the Firefox Accounts keyring.
```

Comment:
I'd change the middle sentence on this line to be `(This might not be the case in Chrome, by the way.)` -- changing `It` to `This` to disambiguate "this idea" rather than the client, the data, or any other previously mentioned nouns, and making the whole idea parenthetical, and using `in` instead of `of` to describe a property found in a piece of software. Also, `keyring` might not be a real term in FxA, just one I made up in storage.sync.